Use Kali packages where possible and use commit hashes where we must install from source

jsf9k opened the pull request:
## 🗣 Description ##

This pull request:
- Modifies the Ansible code to install from Kali system packages vice building from source where possible
- Modifies the Ansible code to install via commit hash where we must build from source
- Adds installation of the `gowitness` Kali package
- Removes the installation of `aquatone`
- Removes the installation of tools based on Python 2

## 💭 Motivation and context ##

- Installing from Kali system packages where possible results in a smaller burden for the maintainers of this repo
- Using a commit hash is more secure than installing whatever code happens to be at the end of a branch or tag
- This package was requested by the assessors
- This package is no longer needed according to the assessors
- Python 2 has been EOL since January 1, 2020

See also cisagov/teamserver-packer#104.

## 🧪 Testing ##

All automated tests pass.  I also built and tested these changes in our `dev-a` and `staging-a` COOL deployments and verified that they function as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.
- [x] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).
- [x] Test new AMI in our `dev-a` COOL environment.
- [x] Create a pre-release (necessary if and only if the pre-release version was bumped).
- [x] Test pre-release in our `staging-a` COOL environment.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release (necessary if and only if the version was bumped).

jsf9k committed:
Bump version from 1.0.3 to 1.0.4-rc.1

jsf9k committed:
Install dirsearch via Kali system package vice building from source

jsf9k committed:
Install lsassy via Kali system package vice building from source

jsf9k committed:
Install mimikatz via Kali system package vice building from source

jsf9k committed:
Install mitm6 via Kali system package vice building from source

jsf9k committed:
Install pypykatz via Kali system package vice building from source

jsf9k committed:
Install SecLists via Kali system package vice building from source

jsf9k committed:
Install sqlmap via Kali system package vice building from source

jsf9k committed:
Install donut via Kali system package vice building from source

jsf9k committed:
Install responder via Kali system package vice building from source

jsf9k committed:
Install sharpshooter via Kali system package vice building from source

jsf9k committed:
Install gowitness via Kali system package

jsf9k committed:
Install bloodhound.py via Kali system package

jsf9k committed:
Bump version from 1.0.4-rc.1 to 1.1.0-rc.1

jsf9k committed:
Remove aquatone

This tool is no longer needed.

jsf9k committed:
Bump version from 1.1.0-rc.1 to 2.0.0-rc.1

jsf9k committed:
Install tarrell13/Auto-Egress-Assess via commit hash

jsf9k committed:
Install 3ndG4me/AutoBlue-MS17-010 via commit hash

jsf9k committed:
Commit C-Sto/BananaPhone via commit hash

jsf9k committed:
Install ajpc500/BOFs via commit hash

jsf9k committed:
Install mdsecactivebreach/CACTUSTORCH via commit hash

jsf9k committed:
Install Techno-Hwizrdry/checkpwnedemails via commit hash

jsf9k committed:
Install trustedsec/CS-Remote-OPs-BOF via commit hash

jsf9k committed:
Install trustedsec/CS-Situational-Awareness-BOF via commit hash

jsf9k committed:
Install bovine/datapipe via commit hash

jsf9k committed:
Install nccgroup/demiguise via commit hash

jsf9k committed:
Install sixdub/DomainTrustExplorer via commit hash

jsf9k committed:
Install tyranid/DotNetToJScript via commit hash

jsf9k committed:
Install MacR6/ftpenum via commit hash

jsf9k committed:
Install GhostPack/Lockless via commit hash

jsf9k committed:
Install GhostPack/Rubeus via commit hash

jsf9k committed:
Install GhostPack/Seatbelt via commit hash

jsf9k committed:
Install GhostPack/SafetyKatz via commit hash

jsf9k committed:
Install GhostPack/SharpDPAPI via commit hash

jsf9k committed:
Install GhostPack/SharpDump via commit hash

jsf9k committed:
Install GhostPack/SharpRoast via commit hash

jsf9k committed:
Install GhostPack/SharpUp via commit hash

jsf9k committed:
Install GhostPack/SharpWMI via commit hash

jsf9k committed:
Install ChrisTruncer/gnmap-parser via commit hash

jsf9k committed:
Install RedSiege/Hasher via commit hash

jsf9k committed:
Install HarmJ0y/ImpDump via commit hash

jsf9k committed:
Install eladshamir/Internal-Monologue via commit hash

jsf9k committed:
Install HarmJ0y/KeeThief via commit hash

jsf9k committed:
Install xan7r/kerberoast via commit hash

jsf9k committed:
Install ChrisTruncer/mikto via commit hash

jsf9k committed:
Install xan7r/Misc via commit hash

jsf9k committed:
Install vysecurity/morphHTA via commit hash

jsf9k committed:
Install worawit/MS17-010 via commit hash

jsf9k committed:
Install noahpowers/nlzr via commit hash

jsf9k committed:
Install PowerShellEmpire/PowerTools via commit hash

jsf9k committed:
Install NetSPI/PowerUpSQL via commit hash

jsf9k committed:
Install xorrior/RandomPS-Scripts via commit hash

jsf9k committed:
Install Arvanaghi/SessionGopher via commit hash

jsf9k committed:
Install mubix/shellshocker-pocs via commit hash

jsf9k committed:
Install SimplySecurity/SimplyEmail via commit hash

jsf9k committed:
Install SimplySecurity/SimplyTemplate via commit hash

jsf9k committed:
Install MacR6/sshenum via commit hash

jsf9k committed:
Install rasta-mouse/TikiTorch via commit hash

jsf9k committed:
Install tomnomnom/waybackurls via commit hash

jsf9k committed:
Install lgandx/PCredz via commit hash

jsf9k committed:
Install frohoff/ysoserial via commit hash

jsf9k committed:
Install RedSiege/EXCELntDonut via commit hash

jsf9k committed:
Remove install of fox-it/Bloodhound.py from source

This is now being installed as a Kali Linux system package.

jsf9k committed:
Install Tylous/ScareCrow via a commit hash

Note that we are also switching from optiv/ScareCrow to
Tylous/ScareCrow per assessor request.

jsf9k committed:
Remove installation of nccgroup/demiguise

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Remove installation of MacR6/ftpenum

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Remove installation of HarmJ0y/ImpDump

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Remove installation of xan7r/kerberoast

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Remove installation of vysecurity/morphHTA

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Remove installation of mubix/shellshocker-pocs

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Remove installation of SimplySecurity/SimplyEmail

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Remove installation of SimplySecurity/SimplyTemplate

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Remove installation of MacR6/sshenum

According to the assessors this Python 2 tool is no longer needed.

jsf9k committed:
Install Accenture/CLRvoyance via commit hash

jsf9k committed:
Install BloodHound as a Kali Linux system package

Note that Node4j requires OpenJDK 11, so we also have to install that.
It is safe to delay this installation until we actually run Neo4j.

Also move the installation of BloodHound.py to bloodhound.py since
that is a better home for it.

jsf9k committed:
Install outflanknl/C2-Tool-Collection via commit hash

jsf9k committed:
Install killswitch-GUI/CobaltStrike-Toolkit via commit hash

jsf9k committed:
Install rsmudge/Malleable-C2-Profiles via commit hash

jsf9k committed:
Install bluscreenofjeff/Malleable-C2-Randomizer via commit hash

dav3r reviewed (approved):
Lock it all down! 🔒 👍

coffeegist commented on ansible/bloodhound.yml:
Let's add `bloodhound-ce-python` to this. `bloodhound.py` is a legacy collector, and the ce python repo is the latest.

jsf9k replied:
Please see commit e5affa9.  I built a new `dev-a` COOL AMI with this change and verified that it functions as expected.

coffeegist commented on ansible/bloodhound.yml:
```diff
@@ -1,162 +1,43 @@
-      ansible.builtin.package:
-        name:
-          - neo4j
+    - name: Start Neo4j
```
Neo4j is installed by bloodhound's kali package. See http://kali.org/tools/bloodhound/ for more details. Looks like we'd still need to change the initial neo4j password the way we currently do AND also set that password in `/etc/bhapi/bhapi.json`

jsf9k replied:
Please see commit 72aac93.  I built a new `dev-a` COOL AMI with this change and verified that it functions as expected.  I was able to start up BloodHound.

The only complaints I noticed were about the missing directories `/etc/bloodhound/collectors/sharphound` and `/etc/bloodhound/collectors/azurehound`.  Do you need me to pre-create those two directories, or will they be created when/if you run `sharphound` and/or `azurehound`?

coffeegist replied:
I'm not sure about that, I haven't used it. But we wouldn't run sharphound from kali so I think we'll be ok without them. If we need to add them in, we can do so in the future

jsf9k replied:
Are we good to resolve this conversation and merge the PR?

mcdonnnj commented on ansible/assessor_tools.yml:
I know it's annoying, but for any of these commit hashes that align with a tag/release it would be great to have an attribution comment. Something like:

```yaml
        assessment_tool_archive_src: https://github.com/bovine/datapipe/tarball/31ab7b9702f377ad5c2b87326b647670e83d1c5f  # v1.0
```

jsf9k replied:
I don't think any of them align with a tag or release.  I think they are just the current end of the main branch.  I will check with @coffeegist to make sure.

jsf9k replied:
@coffeegist confirmed my suspicion.  Can we resolve this conversation, or do you want attribution comments stating the default branch name?

mcdonnnj replied:
No, that's fine. If they just want the latest commit on the default branch then it is what it is. If there are any that diverge from that (non-default branch) it'd be nice to mention that, but that's all.

jsf9k replied:
There were none that diverged from that.

mcdonnnj reviewed (approved):
This is a great change 💪💪💪 I have one mildly annoying suggestion for your consideration.

jsf9k committed:
Finalize version from 2.0.0-rc.1 to 2.0.0

jsf9k committed:
Add bloodhound-ce-python

This was requested by @coffeegist since bloodhound.py is a legacy
collector and bloodhound-ce-python is the latest.

Co-authored-by: Adam <1392689+coffeegist@users.noreply.github.com>

jsf9k committed:
Add code to set the Neo4j password in the BH API config file

The Neo4j password must also be set here.

Co-authored-by: Adam <1392689+coffeegist@users.noreply.github.com>

dav3r reviewed (approved):
Re-approved with gusto 👏
